pkgs/swiftgen/swift2objc/lib/src/transformer/transformers/transform_method.dart

```diff
@@ -62,23 +69,54 @@ List<String> _generateMethodStatements(
     arguments.add(methodCallArg);
   }
 
+  final methodReturnType = originalMethod.returnType;
+
   final originalMethodCall =
-      "let result = ${wrappedClassInstance.name}.${originalMethod.name}(${arguments.join(", ")})";
+      "${wrappedClassInstance.name}.${originalMethod.name}(${arguments.join(", ")})";
+
+  if (methodReturnType.isObjCRepresentable) {
+    return ["return $originalMethodCall"];
+  }
+
+  if (methodReturnType is GenericType) {
+    throw UnimplementedError("Generic types are not implemented yet");
+  }
+
+  final transformedReturnTypeDeclaration = transformDeclaration(
+    (methodReturnType as DeclaredType).declaration,
+    globalNamer,
+    transformationMap,
+  );
+
+  if (transformedReturnTypeDeclaration is! ClassDeclaration) {
+    throw "A method call result can only be wrapped in a class";
```

Comment:
Don't throw strings. This might be better as an `assert` anyway. Asserts are easier to write and are only evaluated in debug mode, so they're good for checks like this, which are just checking for mistakes in your own code.